sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -163,6 +164,63 @@ private[joins] object HashedRelation {
       new GeneralHashedRelation(hashTable)
     }
   }
+
+  /**
+   * Consume a `LocalNode` to create a `HashedRelation`. Because `LocalNode` is not an `Iterator`,
+   * we cannot use `apply` directly. Moreover, to avoid creating another layer of `Iterator`, we
+   * have to duplicate most codes of `apply` here.
+   *
+   * Note: the default parameter is in conflict with overloading. So it uses a different method than
+   * rather than `apply`.
```

Comment:
Note: this method must be called something other than `apply` because of default parameter overloading conflicts.